src/x/sync/pooled_worker_pool_test.go

```diff
@@ -54,11 +58,13 @@ func TestPooledWorkerPoolGoWithTimeout(t *testing.T) {
 		workers         = 2
 		channelCapacity = 1
 	)
-	// So we can control how empty the worker pool chanel is we
+	// So we can control how empty the worker pool channel is we
 	// set capacity to be same as num workers.
+	disableParallelExecutionLock.Lock()
 	pooledWorkerPoolGoroutinesCapacity = &channelCapacity
 	defer func() {
 		pooledWorkerPoolGoroutinesCapacity = nil
+		disableParallelExecutionLock.Unlock()
```

Comment:
Assuming this is just a fix for a flakey test yeah?

Reply:
Yeah that's right. It is using a global var so we need to lock around it